collector/collector.cpp

```diff
@@ -107,9 +107,15 @@ int InsertModule(int fd, const std::unordered_map<std::string, std::string>& arg
     args_str += entry.first + "=" + entry.second;
   }
   CLOG(DEBUG) << "Kernel module arguments: " << args_str;
-  int res = finit_module(fd, args_str.c_str(), 0);
-  if (res != 0) return res;
   struct stat st;
+  fstat(fd, &st);
```

Comment:
We should probably check the return code for `fstat`, just to be extra sure we are not trying to push bogus information into the kernel.